subnets.tf

```diff
@@ -43,6 +43,6 @@ resource "aws_nat_gateway" "nat_gws" {
   for_each = toset(var.private_subnet_cidr_blocks)
 
   allocation_id = aws_eip.nat_gw_eips[each.value].id
-  subnet_id     = module.private.subnets[each.value].id
+  subnet_id     = module.public.subnets[var.public_subnet_cidr_blocks[index(var.private_subnet_cidr_blocks, each.value)]].id
```

Comment:
This line is now a bit of a brain-twister, at least for me- it took me a minute to grep what was going on.  Can you please add a comment here to assist future readers?

Reply:
This was something I intended to do but forgot about.  See commit 3d80193.